lib/semantics/semantics.cc

```diff
@@ -228,6 +230,45 @@ void DoDumpSymbols(std::ostream &os, const Scope &scope, int indent) {
   --indent;
 }
 
+void Semantics::DumpSymbolsSources(std::ostream &os) {
+  std::list<Symbol> symbols;
+  auto less = [&](Symbol a, Symbol b) {
+    return a.name().begin() < b.name().begin();
+  };
+  std::set<Symbol, decltype(less)> visited(less);
+  GetAllSymbols(os, context_.globalScope(), cooked_, symbols);
+  for (auto symbol : symbols) {
+    auto sourceInfo = cooked_.GetLineAndColumnsFromCooked(symbol.name());
+    if (visited.find(symbol) != visited.end()) {
+      continue;
+    }
+    if (sourceInfo) {
+      os << symbol.name().ToString() << ": " << std::get<0>(*sourceInfo) << ","
+         << std::get<1>(*sourceInfo) << "," << std::get<0>(*sourceInfo) << ","
+         << std::get<2>(*sourceInfo) << "\n";
+      visited.emplace(symbol);
+    } else if (const auto useDetails{
+                   symbol.detailsIf<semantics::UseDetails>()}) {
+      os << symbol.name().ToString() << ": "
+         << useDetails->rootModule().name().ToString() << "\n";
+      visited.emplace(symbol);
+    }
+  }
+}
+
+void GetAllSymbols(std::ostream &os, const Scope &scope,
+    const parser::CookedSource &cooked, std::list<Symbol> &symbols) {
```

Comment:
If you use `std::set` instead of `std::list` you won't need `visited` up above, right?